src/legacy/core_plugins/kibana/public/discover/plugin.ts

```diff
@@ -156,22 +159,31 @@ export class DiscoverPlugin implements Plugin<DiscoverSetup, DiscoverStart> {
    */
   private async registerEmbeddable(core: CoreStart, plugins: DiscoverStartPlugins) {
     const { SearchEmbeddableFactory } = await import('./np_ready/embeddable');
-    const getInjector = async () => {
-      if (!this.initializeServices) {
-        throw Error('Discover plugin registerEmbeddable:  initializeServices is undefined');
-      }
-      await this.initializeServices();
-      getInnerAngularModuleEmbeddable(embeddableAngularName, core, plugins);
-      const mountpoint = document.createElement('div');
-      return angular.bootstrap(mountpoint, [embeddableAngularName]);
-    };
     const isEditable = () => core.application.capabilities.discover.save as boolean;
 
+    if (!this.getEmbeddableInjector) {
+      throw Error('Discover plugin method getEmbeddableInjector is undefined');
+    }
+
     const factory = new SearchEmbeddableFactory(
       plugins.uiActions.executeTriggerActions,
-      getInjector,
+      this.getEmbeddableInjector,
       isEditable
     );
     plugins.embeddable.registerEmbeddableFactory(factory.type, factory);
   }
+
+  private async getInjector() {
+    if (!this.embeddableInjector) {
+      if (!this.initializeServices) {
+        throw Error('Discover plugin getEmbeddableInjector:  initializeServices is undefined');
+      }
+      const { core, plugins } = await this.initializeServices();
```

Comment:
@flash I modified `initializeServices` a bit. It's kind of a hack, just to not call and wait for `coreSetup.getStartServices`, since `initializeServices` has already had `core` and `plugins` in its context. I'm not sure if it's applicable from the architecture prospective.

Reply:
I'm fine with that solution!